src/cmd/ksh93/tests/bracket.sh

```diff
@@ -471,4 +471,7 @@ x=10
 
 ([[ x -eq 10 ]]) 2> /dev/null || err_exit 'x -eq 10 fails in [[...]] with x=10'
 
+# POSIX specifies that on error, test builtin should always return value > 1
+test 123 -eq 123x 2>/dev/null || [[ $? -ge 2 ]] || err_exit 'test builtin should return value greater than 1 on error'
```

Comment:
I'm not a fan of chaining `||` and `&&`. It's also not needed in this case and just makes it harder to understand the logic. Just

```
test  ...
[[ $? -ge 2 ]] || err_exit
```